examples/chef/common/clusters/channel/ChannelManager.cpp

```diff
@@ -0,0 +1,343 @@
+    bool nameMatch = channel.name.HasValue() ? channel.name.Value().data_equal(match) : false;
+    bool affiliateCallSignMatch =
+        channel.affiliateCallSign.HasValue() ? channel.affiliateCallSign.Value().data_equal(match) : false;
+    bool callSignMatch = channel.callSign.HasValue() ? channel.callSign.Value().data_equal(match) : false;
+    bool numberMatch   = match.data_equal(chip::CharSpan::fromCharString(number));
+
+    return affiliateCallSignMatch || callSignMatch || nameMatch || numberMatch;
+}
+
+void ChannelManager::HandleChangeChannel(CommandResponseHelper<ChangeChannelResponseType> & helper, const CharSpan & match)
+{
+    std::vector<ChannelInfoType> matchedChannels;
+    uint16_t index = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if CharSpan matches channel name
+        // or callSign or affiliateCallSign or majorNumber.minorNumber
+        if (isChannelMatched(channel, match))
+        {
+            matchedChannels.push_back(channel);
+        }
+        else if (matchedChannels.size() == 0)
+        {
+            // "index" is only used when we end up with matchedChannels.size() == 1.
+            // In that case, we want it to be the number of non-matching channels we saw before
+            // the matching one.
+            index++;
+        }
+    }
+
+    ChangeChannelResponseType response;
+
+    // Error: Found multiple matches
+    if (matchedChannels.size() > 1)
+    {
+        response.status = chip::app::Clusters::Channel::StatusEnum::kMultipleMatches;
+        helper.Success(response);
+    }
+    else if (matchedChannels.size() == 0)
+    {
+        // Error: Found no match
+        response.status = chip::app::Clusters::Channel::StatusEnum::kNoMatches;
+        helper.Success(response);
+    }
+    else
+    {
+        response.status      = chip::app::Clusters::Channel::StatusEnum::kSuccess;
+        response.data        = chip::MakeOptional(CharSpan::fromCharString("data response"));
+        mCurrentChannel      = matchedChannels[0];
+        mCurrentChannelIndex = index;
+        helper.Success(response);
+    }
+}
+
+bool ChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, const uint16_t & minorNumber)
+{
+    bool channelChanged = false;
+    uint16_t index      = 0;
+    for (auto const & channel : mChannels)
+    {
+        // verify if major & minor matches one of the channel from the list
+        if (channel.minorNumber == minorNumber && channel.majorNumber == majorNumber)
+        {
+            // verify if channel changed by comparing values of current channel with the requested channel
+            if (channel.minorNumber != mCurrentChannel.minorNumber || channel.majorNumber != mCurrentChannel.majorNumber)
+            {
+                channelChanged       = true;
+                mCurrentChannelIndex = index;
+                mCurrentChannel      = channel;
+            }
+        }
```

Comment:
you can early-return true right away here (or do we expect multiple matches?)

Reply:
done